analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs

```diff
@@ -100,47 +101,46 @@ public void SetParameterValues_WhenGivenValidSonarLintFileAndDoesNotContainAnaly
         }
 
         [TestMethod]
-        public void SetParameterValues_CalledTwiceAfterChangeInConfigFile_UpdatesProperties()
+        public void SetParameterValues_WithNonExistingPath_UsesInMemoryText()
         {
             // Arrange
-            var sonarLintXmlRelativePath = "ResourceTests\\ToChange\\SonarLint.xml";
-            var options = TestHelper.CreateOptions(sonarLintXmlRelativePath);
+            var inMemoryText = SourceText.From(File.ReadAllText("ResourceTests\\ToChange\\SonarLint.xml"));
+            var options = TestHelper.CreateOptions("ResourceTests\\ThisPathDoesNotExist\\SonarLint.xml", inMemoryText);
             var analyzer = new ExpressionComplexity(); // Cannot use mock because we use reflection to find properties.
 
-            // Act with the file on disk
-            ParameterLoader.SetParameterValues(analyzer, options);
-            analyzer.Maximum.Should().Be(1); // from the XML on disk
-
-            // Update the file on disk
-            var originalContent = File.ReadAllText(sonarLintXmlRelativePath);
-            var modifiedContent = originalContent.Replace("<Value>1</Value>", "<Value>42</Value>");
-            File.WriteAllText(sonarLintXmlRelativePath, modifiedContent);
+            // Act
             ParameterLoader.SetParameterValues(analyzer, options);
-            analyzer.Maximum.Should().Be(42);
 
-            // revert change
-            File.WriteAllText(sonarLintXmlRelativePath, originalContent);
+            // Assert
+            analyzer.Maximum.Should().Be(1); // In-memory value
         }
 
         [TestMethod]
-        public void SetParameterValues_WithMalformedXml_DoesNotPopulateProperties()
+        public void SetParameterValues_CalledTwiceAfterChangeInConfigFile_UpdatesProperties()
```

Comment:
It might be neater just to inline the XML here and in `SetParameterValues_WithNonExistingPath_UsesInMemoryText`, like you did for the test in the first commit.